fake_spectra/spectra.py

```diff
@@ -378,11 +379,27 @@ def load_savefile(self,savefile=None):
         self.axis = np.array(grp["axis"])
         f.close()
 
-    def _interpolate_single_file(self,nsegment, elem, ion, ll, get_tau):
+    def _interpolate_single_file(self,nsegment, elem, ion, ll, get_tau, arepo=False):
         """Read arrays and perform interpolation for a single file"""
-        (pos, vel, elem_den, temp, hh, amumass) = self._read_particle_data(nsegment, elem, ion,get_tau)
-        if amumass is False:
-            return np.zeros([np.shape(self.cofm)[0],self.nbins],dtype=np.float32)
+        if arepo:
+            for nseg in range(self.snapshot_set.get_n_segments()):
+                (pos_, vel_, elem_den_, temp_, hh_, amumass) = self._read_particle_data(nseg, elem, ion, get_tau)
+                if amumass is False:
+                    return np.zeros([np.shape(self.cofm)[0],self.nbins],dtype=np.float32)
```

Comment:
This is the problem: this should be 'continue'. Otherwise it does not correctly compute the particle arrays